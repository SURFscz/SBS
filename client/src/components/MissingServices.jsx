import React from "react";
import I18n from "i18n-js";
import "./MissingServices.scss";
import DOMPurify from "dompurify";

export default function MissingServices({nbrServices}) {

    return (
        <div className={`missing-services services_${nbrServices}`}>
            <span dangerouslySetInnerHTML={{__html: DOMPurify.sanitize(I18n.t("models.services.missingServices"))}}/>
        </div>
    );

}

