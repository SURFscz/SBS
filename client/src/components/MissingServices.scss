@import "../stylesheets/vars";


div.missing-services {
    width: 100%;
    max-width: $medium;

    @media (max-width: $compact-medium) {
        padding: 0 15px;
    }

    margin: 20px auto;
    padding: 20px;
    background-color: #ffeed9;

    @for $i from 0 through 3 {
        &.services_#{$i} {
            margin-top: calc(250px - (#{$i} * 75px));
        }
    }

    a {
        font-weight: bold;
    }
}
