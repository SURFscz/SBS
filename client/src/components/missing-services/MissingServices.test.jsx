import React from "react";
import { render } from "@testing-library/react";
import MissingServices from "./MissingServices";

jest.mock("../../locale/I18n", () => ({
  t: (key, opts) => key + (opts ? JSON.stringify(opts) : ""),
}));

describe("MissingServices", () => {
    it("matches snapshot with default props", () => {
        const { asFragment } = render(
            <MissingServices/>
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("matches snapshot with custom message", () => {
        const { asFragment } = render(
            <MissingServices message="Custom message"/>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
