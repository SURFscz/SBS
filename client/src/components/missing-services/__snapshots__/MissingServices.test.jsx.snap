// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissingServices matches snapshot with custom message 1`] = `
<DocumentFragment>
  <div
    class="missing-services services_undefined"
  >
    <span>
      models.services.missingServices
    </span>
  </div>
</DocumentFragment>
`;

exports[`MissingServices matches snapshot with default props 1`] = `
<DocumentFragment>
  <div
    class="missing-services services_undefined"
  >
    <span>
      models.services.missingServices
    </span>
  </div>
</DocumentFragment>
`;
